### PR TITLE
Upgrade RDFLib requirement

### PR DESCRIPTION
**Pull request by amercader**
The current version of RDFlib won't work on Python 3.9 so we need to upgrade to RDFlib 6, which is not compatible with Python 2. The JSON-LD requirement has been integrated into the main rdflib package.

This adapts the tests so they can run across all RDFLib / Python / CKAN version :crossed_fingers:

### CODE REVIEW

**Comment by amercader** [comment]
I finally managed to get all versions green :scream_cat: 
@seitenbau-govdata (and others)  I'm fairly confident that the RDFlib upgrade won't break existing serializations and parsers but I'd be supergrateful if this PR could be tested in a real life instance with real metadata.

**Comment by seitenbau-govdata** [comment]
@amercader Sure, we can do that. We can test it in the environments CKAN 2.8 under Python 2.7 and CKAN 2.9 under Python 3.6.

**Comment by seitenbau-govdata** [comment]
@amercader Our environment is currently locked to Python 3.6. Unfortunately RDFlib 6 requires Python 3.7+, which means we cannot test this at the moment.

**Comment by amercader** [comment]
@seitenbau-govdata It's a tricky balance. I know that Ubuntu 18.04 might be supported for a while, but Python 3.6 is no longer supported and doesn't receive security updates any more. I'm keen on maximizing support for this extension but also using recent versions to minimize issues like #205.

If I understand correctly, users running on Python 3.6 can still use the extension if they install the `requirements-py2.txt` ones right? Obviously we need to document that on the README (and maybe rename the file to `requirements-py2-py36.txt`) but do you think this would be reasonable?

**Comment by seitenbau-govdata** [comment]
@amercader True, it's really tricky. Yes, the requirements in `requirements-py2.txt` still working under Python 3.6. I think this is a solution we can go with. Adding a hint in the documentation sounds good to me. Probably renaming the requirements file to `requirements-py2-py36.txt`makes it even clearer than just documenting it on the README.

Currently we are testing the branch with the dependencies `rdflib==5.0.0` and `rdflib-jsonld==0.5.0` under CKAN 2.9 (Python 3.6) and CKAN 2.8 (Python 2.7). It wouldn't hurt to update the dependencies at the same time.

File: dev-requirements-py2.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+responses==0.17.0
```
Comment (seitenbau-govdata) [inline dev-requirements-py2.txt:1]:
```suggestion
responses==0.10.15
```

File: requirements-py2.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+rdflib==4.2.1
```
Comment (seitenbau-govdata) [inline requirements-py2.txt:1]:
```suggestion
rdflib==5.0.0
```

File: requirements-py2.txt [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+rdflib-jsonld==0.4.0
```
Comment (seitenbau-govdata) [inline requirements-py2.txt:2]:
```suggestion
rdflib-jsonld==0.5.0
```

**Comment by seitenbau-govdata** [comment]
@amercader And we need a rebase to the current master and adding the `self._add_responses_solr_passthru()` to the new tests
* test_harvest_default_file_size
* test_harvest_config_file_size

**Comment by seitenbau-govdata** [comment]
@amercader 👍 LGTM, thanks!

**Comment by amercader** [comment]
Thanks @seitenbau-govdata ! I've incorporated all your suggested changes, I think that gives a good support across versions, I'll release a new version with this and your other recent changes.

**Pull request merged**